go.sum

```diff
@@ -63,6 +63,8 @@ github.com/Azure/go-ntlmssp v0.0.0-20220621081337-cb9428e4ac1e h1:NeAW1fUYUEWhft
 github.com/Azure/go-ntlmssp v0.0.0-20220621081337-cb9428e4ac1e/go.mod h1:chxPXzSsl7ZWRAuOIE23GDNzjWuZquvFlgA8xmpunjU=
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
 github.com/BurntSushi/xgb v0.0.0-20160522181843-27f122750802/go.mod h1:IVnqGOEym/WlBOVXweHU+Q+/VP0lqqI8lqeDx9IjBqo=
+github.com/ClifHouck/grpc-go-pool v0.1.0 h1:YlG2qlfB/eEfM87Jv1FxfEeiuNiDFyABla0rRR4qFI4=
```

Comment:
should this be referencing your github?

Reply:
+1

Reply:
Depends. The root of my fork does not seem to be actively maintained. I've made modifications to the pool that make more sense for grpc connections. For instance, grpc connections are safe to use across many gorountines at once, so a client grabbing one shouldn't remove it from the pool. The pool just round-robins access to clients to load balance their use.

Reply:
As a OSS team I think the right approach would be to open this issue on their repo and maybe propose a fix to their code.  

I don't think forking and including your fork is a pattern we want to encourage from the open source team.

Reply:
Generally I agree with you, but the context of this repo matters. 

I'm happy to move this fork into `armadaproject` or just move the two source files into `armadaproject/armada`. To reiterate: the root of my fork is not actively maintained, as in it hasn't had a commit for *years*. It's likely maintained by a single person who has effectively abandoned the project. The changes I've made here also totally change the semantics and functionality of the pool, so unlikely to be merged upstream.